cannot compile since commit 742bcdb2bcbd2ebdd0b1de4921e31214ba72d0d8

### Issue and Steps to Reproduce
I'm trying to compile from the master branch, but it seems i can no longer compile from source since  commit 742bcdb2bcbd2ebdd0b1de4921e31214ba72d0d8

```
wallet/db_sqlite3.c: In function 'trace_sqlite3':
wallet/db_sqlite3.c:14:17: error: expected identifier or '(' before '=' token
  struct db_stmt = (struct db_stmt*)stmtv;
                 ^
wallet/db_sqlite3.c:15:17: error: passing argument 1 of 'db_changes_add' from incompatible pointer type [-Werror=incompatible-pointer-types]
  db_changes_add(stmt, stmt);
                 ^
In file included from wallet/db_sqlite3.c:1:0:
./wallet/db_common.h:145:6: note: expected 'struct db_stmt *' but argument is of type 'const char *'
 void db_changes_add(struct db_stmt *db_stmt, const char * expanded);
      ^
wallet/db_sqlite3.c: In function 'db_sqlite3_exec':
wallet/db_sqlite3.c:89:16: error: 'db' undeclared (first use in this function)
  sqlite3_trace(db->sql, trace_sqlite3, stmt);
                ^
wallet/db_sqlite3.c:89:16: note: each undeclared identifier is reported only once for each function it appears in
cc1: all warnings being treated as errors
<builtin>: recipe for target 'wallet/db_sqlite3.o' failed
make: *** [wallet/db_sqlite3.o] Error 1
```
Building on ubuntu 16.04.6 LTS

### `getinfo` output
none, cannot compile, so i can't produce getinfo output

same issue here 
gcc (Ubuntu 8.3.0-6ubuntu1) 8.3.0

`wallet/db_sqlite3.c: In function ‘trace_sqlite3’:
wallet/db_sqlite3.c:14:17: error: expected identifier or ‘(’ before ‘=’ token
  struct db_stmt = (struct db_stmt*)stmtv;
                 ^
wallet/db_sqlite3.c:15:17: error: passing argument 1 of ‘db_changes_add’ from incompatible pointer type [-Werror=incompatible-pointer-types]
  db_changes_add(stmt, stmt);
                 ^
In file included from wallet/db_sqlite3.c:1:0:
./wallet/db_common.h:145:6: note: expected ‘struct db_stmt *’ but argument is of type ‘const char *’
 void db_changes_add(struct db_stmt *db_stmt, const char * expanded);
      ^
wallet/db_sqlite3.c: In function ‘db_sqlite3_exec’:
wallet/db_sqlite3.c:89:16: error: ‘db’ undeclared (first use in this function)
  sqlite3_trace(db->sql, trace_sqlite3, stmt);
                ^
wallet/db_sqlite3.c:89:16: note: each undeclared identifier is reported only once for each function it appears in
cc1: all warnings being treated as errors
<builtin>: recipe for target 'wallet/db_sqlite3.o' failed
make: *** [wallet/db_sqlite3.o] Error 1
rm external/libwally-core/src/secp256k1/libsecp256k1.la
`

--- a/wallet/db_sqlite3.c
+++ b/wallet/db_sqlite3.c
@@ -11,8 +11,8 @@
 /* Prior to sqlite3 v3.14, we have to use tracing to dump statements */
 static void trace_sqlite3(void *stmtv, const char *stmt)
 {
-	struct db_stmt = (struct db_stmt*)stmtv;
-	db_changes_add(stmt, stmt);
+	struct db_stmt *s = (struct db_stmt*)stmtv;
+	db_changes_add(s, stmt);
 }
 #endif
 
@@ -86,7 +86,7 @@ static bool db_sqlite3_exec(struct db_stmt *stmt)
 #if !HAVE_SQLITE3_EXPANDED_SQL
 	/* Register the tracing function if we don't have an explicit way of
 	 * expanding the statement. */
-	sqlite3_trace(db->sql, trace_sqlite3, stmt);
+	sqlite3_trace(stmt->db->conn, trace_sqlite3, stmt);
 #endif
 
 	if (!db_sqlite3_query(stmt)) {
@@ -110,7 +110,7 @@ static bool db_sqlite3_exec(struct db_stmt *stmt)
 #else
 	/* Unregister the trace callback to avoid it accessing the potentially
 	 * stale pointer to stmt */
-	sqlite3_trace(db->sql, NULL, NULL);
+	sqlite3_trace(stmt->db->conn, NULL, NULL);
 #endif
 
 	return true;

